sklearn/gaussian_process/gpr.py

```diff
@@ -320,13 +322,18 @@ def predict(self, X, return_std=False, return_cov=False):
                 y_cov = self.kernel_(X) - K_trans.dot(v)  # Line 6
                 return y_mean, y_cov
             elif return_std:
-                # compute inverse K_inv of K based on its Cholesky
-                # decomposition L and its inverse L_inv
-                L_inv = solve_triangular(self.L_.T, np.eye(self.L_.shape[0]))
-                K_inv = L_inv.dot(L_inv.T)
+                # cache result of K_inv computation
+                if self._K_inv is None:
+                    # compute inverse K_inv of K based on its Cholesky
+                    # decomposition L and its inverse L_inv
+                    L_inv = solve_triangular(self.L_.T,
+                                             np.eye(self.L_.shape[0]))
+                    self._K_inv = L_inv.dot(L_inv.T)
+
                 # Compute variance of predictive distribution
                 y_var = self.kernel_.diag(X)
-                y_var -= np.einsum("ij,ij->i", np.dot(K_trans, K_inv), K_trans)
+                sum1 = np.dot(K_trans, self._K_inv).T
+                y_var -= np.einsum("ki,ik->k", K_trans, sum1)
```

Comment:
Is there a reason you changed this? I thought this PR was only about caching `self._K_inv`. As @TomDLT  mentioned in https://github.com/scikit-learn/scikit-learn/issues/9234#issuecomment-311408490, the einsum optimization was arleady done in #8591.